__tests__/index.test.js

```diff
@@ -1,12 +1,43 @@
-const requireFromString = require('require-from-string');
 const ejsLoader = require('../index.js');
 
-describe('ejsLoader', function() {
-  it('returns template with applied parameters', function() {
-    const template = '<div>Hello <%= world %>!</div>';
-    const params = { world: 'World' };
-    const compiled = requireFromString(ejsLoader(template));
+const compileTemplate = (template, mockedLoaderContext = {}) => {
+  const mergedMockedLoaderContext = {
+    cacheable: null,
+    ...mockedLoaderContext
+  };
 
+  return ejsLoader.call(mergedMockedLoaderContext, template);
+}
+
+const convertTemplateStringToFunction = (templateString) => {
+  const removeExportDefaultString = templateString.match(/export default (.*)/s)[1];
+  return new Function(`return ${removeExportDefaultString}`)();
+}
+
+describe('ejsLoader', () => {
+  it('returns template with applied parameters with supplied variable option', () => {
+    const template = '<div>Hello <%= data.world %>!</div>';
+    const params = { 
+      world: 'World'
+    };
+    const compilerOptions = {
+      // stub out options populated by Webpack and handled internally by loader-utils
+      query: {
+        variable: 'data'
+      }
+    };
+    const compiled = convertTemplateStringToFunction(compileTemplate(template, compilerOptions));
     expect(compiled(params)).toBe('<div>Hello World!</div>');
   });
-});
+
+  it('throws error when options variable or query variable are undefined', () => {
+    const template = '<div>Hello <%= args.world %>!</div>';
+    expect(() => {
+      compileTemplate(template)
+    }).toThrowError(`
+      To support ES Modules, the 'variable' option must be passed to avoid 'with' statements 
+      in the compiled template to be strict mode compatible. 
+      Please see https://github.com/lodash/lodash/issues/3709#issuecomment-375898111
+    `);
```

Comment:
Consider testing the error type here instead of a message string?

Reply:
@al-the-x I wanted to originally do that but would have to export the error and didn't want to add complexity by exporting an ES6 class. I figured a test on a magic string was the next best option 🤷‍♂️